src/signer.rs

```diff
@@ -84,19 +105,139 @@ impl Signer for FileSigner {
         Ok(signature)
     }
 
-    fn get_public_key(&self) -> anyhow::Result<(PublicKey, [u8; 20])> {
+    async fn get_public_key(&self) -> anyhow::Result<(PublicKey, [u8; 20])> {
         let secp = Secp256k1::new();
         let public_key = self.secret_key.public_key(&secp);
-        let pubkey_uncompressed = public_key.serialize_uncompressed();
-        let pubkey_hash: [u8; 32] = Keccak256::new_with_prefix(&pubkey_uncompressed[1..])
-            .finalize()
-            .into();
-        let pubkey_evm: [u8; 20] =
-            pubkey_hash[pubkey_hash.len() - 20..]
-                .try_into()
-                .map_err(|e| {
-                    anyhow::anyhow!("Failed to convert public key hash to EVM format: {}", e)
-                })?;
+        let pubkey_evm = get_evm_address(&public_key)?;
+        Ok((public_key, pubkey_evm))
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct KMSSigner {
+    client: aws_sdk_kms::Client,
+    arn: aws_arn::ResourceName,
+    public_key: Option<(PublicKey, [u8; 20])>,
```

Comment:
maybe call this cached_public_key :? i was wondering why it could've been optional.